### PR TITLE
Fix named tuple

### PR DESCRIPTION
**Pull request by jasonkhadka**
## Description

Fixes https://github.com/lf1-io/padl/issues/275

   * clean up named_tuple logic, so that final output is always named tuple for Parrallel/Rollout
   * Output is namedtuple for all *apply methods in all modes.
   * Add tests for this logic

### CODE REVIEW

**Comment by wuhu** [comment]
is this related to #276?

File: tests/unittests/test_transforms.py [hunk 1]
```diff
@@ -129,6 +129,53 @@ def test_isinstance_of_namedtuple():
...
+        assert not pd._isinstance_of_namedtuple(self.transform_1.infer_apply(1))
+        assert pd._isinstance_of_namedtuple(self.transform_2.infer_apply(1))
+        assert pd._isinstance_of_namedtuple(self.transform_3.infer_apply(1))
+
+    def test_eval_apply(self):
+        assert not any(list(map(pd._isinstance_of_namedtuple,
+                                self.transform_1.eval_apply([1, 2, 3])
+                                )
+                            )
+                       )
```
Comment (sjrl) [inline tests/unittests/test_transforms.py:149]: Could we put all of these `)` on the same line to save some space. So `)))`

**Comment by sjrl** [comment]
@jasonkhadka Shouldn't we also remove 
```python
        if Transform.pd_mode is not None:
            return tuple(out)
```
from the call methods in `Rollout` and `Parallel` for the time being?

**Comment by sjrl** [comment]
> is this related to #276?

Issue #276 will be a larger refactor so we can used `namedtuple`'s in between transforms. Right now a `namedtuple` can only be returned as the final output of the `_apply` methods. We cannot used `namedtuple`'s everywhere because they cannot be pickled which makes them incompatible with the `DataLoader`.

**Comment by wuhu** [comment]
> Issue #276 will be a larger refactor so we can used namedtuple's in between transforms. Right now a namedtuple can only be returned as the final output of the _apply methods. We cannot used namedtuple's everywhere because they cannot be pickled which makes them incompatible with the DataLoader.

I see, thanks for the explanation.

**Comment by jasonkhadka** [comment]
> @jasonkhadka Shouldn't we also remove
> 
> ```python
>         if Transform.pd_mode is not None:
>             return tuple(out)
> ```
> 
> from the call methods in `Rollout` and `Parallel` for the time being?

this is still needed cause users might still want to call single transform and expect named output. 
example:
```python
t = (a - 'a1' + b - 'b1') >> (c / d)
```
users might do `t[0](arg)` and expect output as `namedtuple('a1': a(arg), 'b1':b(arg))` 
This logic makes it possible.

**Review by sjrl — APPROVED** [review]
Looks good to me!

**Pull request merged**